paasta_tools/kubernetes_tools.py

```diff
@@ -2445,22 +2449,29 @@ def get_pod_template_spec(
 
         # not all services use autoscaling, so we label those that do in order to have
         # prometheus selectively discover/scrape them
-        if self.should_use_metrics_provider(METRICS_PROVIDER_UWSGI):
-            # UWSGI no longer needs a label to indicate it needs to be scraped as all pods are checked for the uwsgi stats port by our centralized uwsgi-exporter
-            # But we do still need deploy_group for relabeling properly
-            # this should probably eventually be made into a default label,
+        metrics_providers_needing_deploy_group = [
+            METRICS_PROVIDER_UWSGI,
+            METRICS_PROVIDER_PISCINA,
+            METRICS_PROVIDER_GUNICORN,
+            METRICS_PROVIDER_WORKER_LOAD,
+        ]
+
+        if any(
+            self.should_use_metrics_provider(provider)
+            for provider in metrics_providers_needing_deploy_group
+        ):
+            # Deploy group is needed for Prometheus relabeling properly
+            # This should probably eventually be made into a default label,
             # but for now we're fine with it being behind these feature toggles.
-            # ideally, we'd also have the docker image here for ease-of-use
+            # Ideally, we'd also have the docker image here for ease-of-use
             # in Prometheus relabeling, but that information is over the
             # character limit for k8s labels (63 chars)
             labels["paasta.yelp.com/deploy_group"] = self.get_deploy_group()
```

Comment:
hmm, maybe i should just make a PR to ship on monday to add this as a default label - it'll require a big bounce, but that's fine :p


(that said, no action from you - if this gets merged before i make my change i can just refactor)